packages/react-html/src/server/components/Script.tsx

```diff
@@ -10,7 +10,7 @@ export function Script(props: Props) {
 
   const attributes = {
     ...otherProps,
-    type: !type || isNoModule ? 'text/javascript' : type,
+    type: isNoModule ? 'text/javascript' : type,
```

Comment:
I don't know why we're forcing `nomodule` to become `text/javascript`, but I won't touch this in this PR.

Reply:
Seems like this logic is just avoiding setting type on module scripts since they already have type=module?